src/Symfony/Component/Ldap/Adapter/ExtLdap/Query.php

```diff
@@ -175,7 +175,6 @@ private function resetPagination()
     {
         $con = $this->connection->getResource();
         $this->controlPagedResultResponse($con, 0, '');
```

Comment:
Looking at the original PR : https://github.com/symfony/symfony/pull/38392/files#diff-24b79f3ac1a99938f5acb158a450e38d30c1984a5d333b5b20f2c38a73d10e31L183-R177

I think the fix, should be to change this line to:
```suggestion
        $this->controlPagedResult($con, 0, '');
```

Reply:
/cc @lucasaba

Reply:
I think it not.... The original error is on #38874 is `Warning: ldap_parse_result() expects parameter 2 to be resource, int given`.

The only place where `ldap_parse_result` receive an in is when it is called from 
`$this->controlPagedResultResponse($con, 0, '');`

I think the correct solutions is to call it with `$this->controlPagedResultResponse($con, null, '');`

Reply:
@lucasaba I think error on #38874 is a result of mistake on #38392 
Why `ldap_control_paged_result` was replaced by `controlPagedResultResponse`, not `controlPagedResult`?

Reply:
my point is, when. you replaced `ldap_parse_result()` by `controlPagedResult` in #38392, in that line you replaced it by `controlPagedResultResponse`.

This explain why the second argument is `0` and not a ressources.

Reply:
I've created a new repository with a test openldap server with the configuration needed by symony's tests.
You can find it here: https://github.com/lucasaba/symfony-ldap-test-server

Just `docker-compose up` it and launch test suite.

@jderusse suggestion fixes the problem. But I need some more time to make other checks. I'm unable to delete entries and I'm not sure if the problem is the LDAP server or the code.

Reply:
By the way: I had to modify AdapterTest and remove

```php
const PAGINATION_REQUIRED_CONFIG = [
        'options' => [
            'protocol_version' => 3,
        ],
    ];
```

And I had to add, in LdapTestCase, after `ldap_connect`, the folloowing:

```php
ldap_set_option($h, LDAP_OPT_PROTOCOL_VERSION, 3);
```

Reply:
@jderusse You're completly right! Its all my bad.
@Nek- I could launch locally the tests and they all pass. I should've done it earlier. I'm not able to add the ldap server on github actions because I still don't have enough knowledge.

To launch the tests, just clone https://github.com/lucasaba/symfony-ldap-test-server, build it and launch it. Then you can run phpunit against the LDAP tests.

Just some more notes (since you're working on it):

- remove definition of the `PAGINATION_REQUIRED_CONFIG` constant from `src/Symfony/Component/Ldap/Tests/Adapter/ExtLdap.php` and all its reference from the same file (it doesn't work)
- add `ldap_set_option($h, LDAP_OPT_PROTOCOL_VERSION, 3);` on `src/Symfony/Component/Ldap/Tests/LdapTestCase.php` just after the connection
- at row 85, in `src/Symfony/Component/Ldap/Tests/Adapter/ExtLdap/LdapManagerTest.php`, modify the test. The exception is a `LdapException`. `AlreadyExistsException` is used for an already existing connection.
- after the `testLdapAddDouble` there is a new entry ('cn=Elsa Amrouche,dc=symfony,dc=com'). This will break all successive tests.... you should find a way to remove the entry just after the test....
- at row 213, in `src/Symfony/Component/Ldap/Tests/Adapter/ExtLdap/LdapManagerTest.php` substitute the line with:
```php
// $newEntry->getAttribute('cn') returns an array not a string
$this->assertContains($originalCN, $newEntry->getAttribute('cn'));
$this->assertContains('Kevin', $newEntry->getAttribute('cn'));
```

Reply:
Idk if it's related to be being under WSL ^^' but even with the docker it's not working (even with rebinding ports).

Since I can't run the test suite I think cleaning the test suite is not a super good idea since it's out of the scope of this PR.